paasta_tools/tron_tools.py

```diff
@@ -605,6 +627,20 @@ def validate(self):
             error_msgs.append(
                 f"{self.get_job_name()}.{self.get_action_name()} must have a deploy_group set"
             )
+        # We are not allowing users to specify `cpus` and `mem` configuration if the action is a Spark job
+        # with driver running on k8s (executor: spark), because we derive these values from `spark.driver.cores`
+        # and `spark.driver.memory` in order to avoid confusion.
+        if self.get_executor() == "spark":
+            if "cpus" in self.config_dict:
+                error_msgs.append(
+                    f"{self.get_job_name()}.{self.get_action_name()} is a Spark job. `cpus` config is not allowed. "
+                    f"Please specify the driver cores using `spark.driver.cores`."
+                )
+            if "mem" in self.config_dict:
+                error_msgs.append(
+                    f"{self.get_job_name()}.{self.get_action_name()} is a Spark job. `mem` config is not allowed. "
+                    f"Please specify the driver memory using `spark.driver.memory`."
+                )
```

Comment:
[optional] we can can add checks in the [yelpsoa_configs' pre-commit hook](https://sourcegraph.yelpcorp.com/sysgit/yelpsoa-configs/-/blob/tron/validate_spark_tronfig), so users can notice the problem earlier before the next run

Reply:
it's probably worth adding this to `paasta validate` now - it'd be essentially dropping this same code in there :)

Reply:
oh actually, this is already being called by `paasta validate` :)